App/Composition/ShowSmilieKeyboardCommand.swift

```diff
@@ -41,6 +53,56 @@ final class ShowSmilieKeyboardCommand: NSObject {
         }
     }
     
+    private func showNewSmiliePicker() {
+        guard var viewController = textView.window?.rootViewController else { return }
+        
+        // Find the topmost presented view controller
+        while let presented = viewController.presentedViewController {
+            viewController = presented
+        }
+        
+        // Check if smilie picker is already being presented
+        if viewController is UIHostingController<SmiliePickerView> {
+            return
+        }
+        
+        // Dismiss keyboard before showing smilie picker
+        textView.resignFirstResponder()
+        
+        weak var weakTextView = textView
+        let pickerView = SmiliePickerView(dataStore: smilieKeyboard.dataStore) { [weak self] smilieData in
+            self?.insertSmilieData(smilieData)
+            // Reactivate keyboard after inserting smilie
+            DispatchQueue.main.async {
```

Comment:
Can you add to the comment why this delay is needed? (I get it, text views, but always worth noting.)